### ISSUE TITLE
backend: request validation returns 500 internal error instead 400 with error description

### ISSUE THREAD

**Issue by RoboSparrow** [gardners/surveysystem#134]
* tests: [invalid_request_returns_400.test](https://github.com/gardners/surveysystem/blob/master/backend/tests/invalid_request_returns_400.test)

example:

```
curl -I http://localhost:3099/surveyapi/nextquestion?sessionid=invalid
```
returns

```
HTTP/1.1 500 Internal Server Error
Content-Type: text/html
Content-Length: 369
Date: Fri, 08 Mar 2019 00:24:45 GMT
Server: lighttpd/1.4.45
```

and logs

```
2019/02/08.10:16.3:src/fcgimain.c:352:fcgi_newsession():Page handler entered.
2019/02/08.10:16.3:src/errorlog.c:54:remember_error():ERROR : surveyid missing from query string
```

expected 

`400 - surveyid missing`, see [fcgi_newsession](https://github.com/gardners/surveysystem/blob/master/backend/src/fcgimain.c#L357)

**Comment by RoboSparrow** [comment]
I add a test for this, pushing that into master and link here

**Comment by RoboSparrow** [comment]
* tests: [invalid_request_returns_400.test](https://github.com/gardners/surveysystem/blob/master/backend/tests/invalid_request_returns_400.test)

**Comment by RoboSparrow** [comment]
closing, merged into #414
